### ISSUE TITLE
Command filter enhancement

### ISSUE THREAD

**Issue by schillingerfgmail** [htop-dev/htop#1146]
I often have processes with really long, several hundred character command lines.  Would it be difficult to add a regex style filter or some other method to extract the more meaningful elements of the command?

**Comment by BenBE** [comment]
Yes, because we currently do not link against a proper regex library.

That being said: The filter currently matches anywhere in the command line.

**Comment by nh2** [comment]
> The filter currently matches anywhere in the command line.

Not "anywhere" if the command line is longer than ~4000 characters:

* https://github.com/htop-dev/htop/issues/1860
